examples/gno.land/r/gnoland/valopers/valopers.gno

```diff
@@ -0,0 +1,455 @@
+// Package valopers is designed around the permissionless lifecycle of valoper profiles.
+package valopers
+
+import (
+	"crypto/bech32"
+	"errors"
+	"regexp"
+	"std"
+	"strconv"
+	"strings"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/avl/pager"
+	"gno.land/p/demo/combinederr"
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/demo/ownable/exts/authorizable"
+)
+
+var (
+	ErrValoperExists       = errors.New("valoper already exists")
+	ErrValoperMissing      = errors.New("valoper does not exist")
+	ErrAddressExists       = errors.New("valoper updated address exists")
+	ErrSameAddress         = errors.New("valoper updated address is the same")
+	ErrInvalidAddress      = errors.New("invalid address")
+	ErrOriginCaller        = errors.New("origin call is not the same")
+	ErrInvalidMoniker      = errors.New("moniker cannot be empty")
+	ErrInvalidDescription  = errors.New("description cannot be empty")
+	ErrInvalidPubKey       = errors.New("invalid public key")
+	ErrInvalidP2PAddress   = errors.New("invalid P2P address")
+	ErrP2PAddressesMissing = errors.New("P2P addresses cannot be empty")
+)
+
+var (
+	valopers     *avl.Tree                  // valopers keeps track of all the active validator operators. Address -> Valoper
+	instructions string                     // markdown instructions for valoper's registration
+	minFee       int64     = 20 * 1_000_000 // minimum gnot must be paid to register.
+	minCoin                = std.NewCoin("ugnot", minFee)
+)
+
+// Valoper represents a validator operator profile
+type Valoper struct {
+	Moniker     string // A human-readable name
+	Description string // A description and details about the valoper
+
+	Address      std.Address // The bech32 gno address of the validator
+	PubKey       string      // The bech32 public key of the validator
+	Active       bool        // Flag indicating if the valoper is active
+	P2PAddresses []string    // The publicly reachable P2P addresses of the validator
+
+	Auth *authorizable.Authorizable    // The authorizer system for the valoper
```

Comment:
GetByAddr is returning a Valoper objet, here, Authorizable is expected to be a safe object, but we need to take care when we add fields; alternatively, you can just unexport everyhibng and create safe getters.

Reply:
Right! I unexported `auth` and made a getter. Thanks.